qualtran/bloqs/block_encoding/block_encoding_base.py

```diff
@@ -67,6 +68,31 @@ class `BlockEncoding` bloq, which expects values for $\alpha$, $\epsilon$,
     def pretty_name(self) -> str:
         return 'B[H]'
 
+    @property
+    def dtype(self) -> QDType:
+        """The type of the system being block encoded."""
+        raise NotImplementedError
+
+    @property
+    def alpha(self) -> SymbolicFloat:
+        """The normalization constant."""
+        raise NotImplementedError
+
+    @property
+    def num_ancillas(self) -> int:
+        """The number of ancilla qubits."""
+        raise NotImplementedError
+
+    @property
+    def num_resource(self) -> int:
+        """The number of resource qubits not counted in ancillas."""
+        raise NotImplementedError
+
+    @property
+    def epsilon(self) -> SymbolicFloat:
+        """The precision to which the block encoding is to be prepared."""
+        raise NotImplementedError
```

Comment:
Make these `abc.abstractmethod` and make the base class derive from `metaclass=abc.ABCMeta` instead of raising `NotImplementedError`

Reply:
My concern is that existing instances of `BlockEncoding` then could not be instantiated unless we changed them now to implement these methods. Do you agree?

Reply:
I will open this issue at merge time.